### PR TITLE
Remove p3d/AppRunner/plugin system

### PR DESCRIPTION
**Pull request by Moguri**
This is for #516 

Tested with:
  * Running test suite
  * Running `bdist_apps` on the asteroids sample with an icon defined

### CODE REVIEW

File: dtool/PandaVersion.pp [hunk 1]
```diff
@@ -48,4 +48,4 @@
 // integers, but by convention it will consist of four integers, with
 // the first three matching the plugin version, and the fourth integer
 // being incremented with each new Core API revision.
-#define P3D_COREAPI_VERSION $[P3D_PLUGIN_VERSION] 2
+#define P3D_COREAPI_VERSION
```
Comment (rdb) [inline dtool/PandaVersion.pp]: What's the rationale for not removing these lines altogether?

I think the only thing using this is `ParsePluginVersion` in `makepandacore.py`, which is used by `makepanda/getversion.py --runtime`, but its `--runtime` handling should be removed, so then we can also remove `ParsePluginVersion` and `ParseCoreapiVersion` from `makepandacore.py`.

File: direct/src/showbase/AppRunnerGlobal.py [hunk 2]
```diff
@@ -1,11 +0,0 @@
...
-object, which only exists when we are running a .p3d file via
-runp3d.py or via the Panda3D plugin or standalone executable.
-
-This is needed for apps that start themselves by importing
-DirectStart; it provides a place for these apps to look for
-the AppRunner at startup. """
-
-#: Contains the global AppRunner instance, or None if this application
-#: was not run from the runtime environment.
-appRunner = None
```
Comment (rdb) [inline direct/src/showbase/AppRunnerGlobal.py]: Could we leave this module in existence for now, but print a deprecation warning when trying to import it?

File: dtool/PandaVersion.pp (hunk 1 above)
Comment (Moguri) [reply dtool/PandaVersion.pp]: I was thinking it was used by PandaSystem, but it's not so I'l remove them.

File: direct/src/showbase/AppRunnerGlobal.py (hunk 2 above)
Comment (Moguri) [reply direct/src/showbase/AppRunnerGlobal.py]: Do we have a mechanism for doing deprecation, or should I just print a warning if this module is imported?

Comment (rdb) [reply direct/src/showbase/AppRunnerGlobal.py]: Let's just `print()` a warning, protected by `if __debug__:`

File: dtool/PandaVersion.pp (hunk 1 above)
Comment (rdb) [reply dtool/PandaVersion.pp]: We can change the methods on PandaSystem to do whatever they do if the symbols aren't defined.  We should also mark those methods as deprecated.

Comment (Moguri) [reply dtool/PandaVersion.pp]: How do we deprecate C++ code?

Comment (rdb) [reply dtool/PandaVersion.pp]: At the very least put a `@deprecated` tag in the docstring like so:
```cpp
/**
 * Method that does something.
 *
 * @deprecated The runtime system has been removed, this now does nothing.
 */
void PandaSystem::
do_something() {
```

Adding a warning using `std::cerr << "Using deprecated PandaSystem.do_something()\n";` under an `#ifndef NDEBUG` would be appropriate if we want to discourage people even more strongly.  (The prc/logging stuff hasn't been defined yet in dtoolbase, so we can't rely on `nassert_raise` et al.)

File: dtool/PandaVersion.pp [hunk 3]
```diff
@@ -31,21 +31,3 @@
 // build which you will be using to produce a distributable Panda3D
 // package, you should set this string appropriately.
 #define PANDA_PACKAGE_VERSION
```
Comment (rdb) [inline dtool/PandaVersion.pp]: This can also be removed

File: dtool/PandaVersion.pp (hunk 1 above)
Comment (Moguri) [reply dtool/PandaVersion.pp]: I have the `@deprecated` tag in the docstrings. I can also add the `cerr` prints if you'd like.

**Comment by Moguri** [comment]
Any other changes or testing you'd like to see?

**Comment by rdb** [comment]
There are still references to the runtime in installpanda.  Also, the docstrings should be wrapped along the ruler per our coding style.  However, I'll fix these before merging.  Thanks!

**Pull request closed without merging**